core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -253,11 +292,24 @@ private static HttpResponse earlyFailedResponse(Throwable t, ClientRequestContex
         return HttpResponse.ofFailure(cause);
     }
 
+    private static HttpResponse earlyFailedResponse(Throwable t,
+                                                    ClientRequestContext ctx,
+                                                    DecodedHttpResponse response) {
+        final UnprocessedRequestException cause = UnprocessedRequestException.of(t);
+        ctx.cancel(cause);
+        response.close(cause);
```

Comment:
nit; `ctx.cancel` calls `response.close` since the cancellation task is updated. The current implementation is fine though.